.github/workflows/main.yml

```diff
@@ -29,7 +29,13 @@ jobs:
     strategy:
       max-parallel: 4
       matrix:
+<<<<<<< HEAD
         python-version: ["3.8", "3.10"]
+||||||| 5c6e9fd
+        python-version: [3.7, 3.8]
+=======
+        python-version: [3.7, 3.8, 3.9]
+>>>>>>> origin/trapping_extended
```

Comment:
So what happened here is that
- I bumped the minimum version to 3.8, as python 3.7 reached [end of life](https://devguide.python.org/versions/)
- I bumped the maximum version to the version of python to the max version in "security" maintenance support.
- I changed the type from numeric to string, because YAML be like that.


I didn't want to test intermediate versions for CI reasons, but maybe its better to.  I could go either way.

This is a case where I'd recommend just going with HEAD.  Thoughts?